client/state/themes/selectors.js

```diff
@@ -0,0 +1,200 @@
+/** @ssr-ready **/
+
+/**
+ * External dependencies
+ */
+import { some, split } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import { getSiteSlug, getSiteOption, isJetpackSite } from 'state/sites/selectors';
+import { getSitePurchases } from 'state/purchases/selectors';
+import { isPremiumTheme, oldShowcaseUrl } from './utils';
+
+/**
+ * Returns the URL for a given theme's details sheet.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Object}  theme  Theme object
+ * @param  {?Number} siteId Site ID to optionally use as context
+ * @return {?String}        Theme details sheet URL
+ */
+export function getThemeDetailsUrl( state, theme, siteId ) {
+	if ( ! theme ) {
+		return null;
+	}
+
+	if ( isJetpackSite( state, siteId ) ) {
+		return getSiteOption( state, siteId, 'admin_url' ) + 'themes.php?theme=' + theme.id;
+	}
+
+	let baseUrl = oldShowcaseUrl + theme.id;
```

Comment:
What do you think about using template literals instead of string concatenation here?